src/framework/core/js/FluidViewDebugging.js

```diff
@@ -0,0 +1,649 @@
+        }
+    });
+    
+    fluid.debug.highlighter.renderRoot = function (that) {
+        var highlightRoot = $(that.options.markup.highlightRoot);
+        that.container.append(highlightRoot);
+        highlightRoot.click(that.events.highlightClick.fire);
+    };
+    
+    fluid.debug.highlighter.clear = function (highlightRoot) {
+        highlightRoot.empty();
+    };
+    
+    fluid.debug.highlighter.positionProps = ["width","height","marginLeft","marginTop","paddingLeft","paddingTop"];
+    
+    fluid.debug.highlighter.colours = {
+        components: [
+            [0, 0, 0],    // black
+            [255, 0, 0],  // red
+            [255, 255, 0] // yellow
+        ],
+        domBinder: [0, 255, 0],  // green
+        renderer:  [0, 255, 255] // cyan
+    };
+    
+    fluid.debug.arrayToRGBA = function (array) {
+        return "rgba(" + array.join(", ") + ")";
+    };
+    
+    fluid.debug.assignColour = function (colour, alpha) {
+        return [colour[0], colour[1], colour[2], alpha];
+    };
+    
+    fluid.debug.highlighter.indexToColour = function (i, isDomBind, isRenderer) {
+        var a = fluid.debug.assignColour, c = fluid.debug.highlighter.colours.components;
+        var base;
+        if (isRenderer) {
+            base = a(fluid.debug.highlighter.colours.renderer, 0.5);
+        } else if (isDomBind) {
+            base = a(fluid.debug.highlighter.colours.domBinder, 0.5);
+        } else {
+            base = a(c[i % c.length], i > c.length ? 0.2 : 0.5);
+        }
+        return base;
+    };
+    
+    fluid.debug.isRendererSelector = function (component, selectorName) {
+        var isRendererComponent = fluid.componentHasGrade(component, "fluid.rendererComponent");
+        var ignoreContains = fluid.contains(component.options.selectorsToIgnore, selectorName);
+
+        return isRendererComponent ? (!selectorName || ignoreContains ? false : true) : false;
+    };
+    
+    fluid.debug.highlighter.disposeEntries = function (entries, domIds) {
+        return fluid.transform(entries, function (entry, i) {
+            var component = entry.component;
+            var container = component.container;
+            var element = fluid.jById(domIds[i], container[0].ownerDocument);
+            var selectorName = entry.selectorName;
+            var isRendererSelector = fluid.debug.isRendererSelector(component, selectorName);
+            var noHighlight = container.is("body");
+            return {
+                component: component,
+                container: element,
+                noHighlight: noHighlight,
+                selectorName: selectorName,
+                colour: fluid.debug.highlighter.indexToColour(i, selectorName, isRendererSelector)
+            };
+        });
+    };
+
+    fluid.debug.domIdtoHighlightId = function (domId) {
+        return "highlight-for:" + domId;
+    };
+    
+    fluid.debug.highlighter.highlight = function (that, highlightRoot, dispositions) {
```

Comment:
"One function, one responsibility."

This could benefit from being split up into separate functions for creating, styling, and scrolling the highlight element.